src/unixTerminal.ts

```diff
@@ -73,10 +75,12 @@ export class UnixTerminal extends Terminal {
     };
 
     // fork
-    const term = pty.fork(file, args, parsedEnv, cwd, cols, rows, uid, gid, onexit);
+    const term = pty.fork(file, args, parsedEnv, cwd, cols, rows, uid, gid, (encoding === 'utf8'), onexit);
 
     this.socket = new PipeSocket(term.fd);
-    this.socket.setEncoding('utf8');
+    if (encoding !== null) {
+      this.socket.setEncoding(encoding);
```

Comment:
Thanks @rprichard, in that case it might make sense to add a `console.warn` when creating a `WindowsTerminal` with `encoding` specified.